core/src/avm2/globals/flash/utils/bytearray.rs

```diff
@@ -0,0 +1,936 @@
+    if let Some(Value::Object(second_array)) = args.get(0) {
+        let combining_bytes = match second_array.as_bytearray() {
+            Some(b) => b.bytes().clone(),
+            None => return Err("ArgumentError: Parameter must be a bytearray".into()),
+        };
+
+        let offset = args
+            .get(1)
+            .unwrap_or(&Value::Unsigned(0))
+            .coerce_to_u32(activation)? as usize;
+        let length = args
+            .get(2)
+            .unwrap_or(&Value::Unsigned(0))
+            .coerce_to_u32(activation)? as usize;
+
+        // In the docs it says "If offset or length is out of range, they are clamped to the beginning and end of the bytes array."
+        // However, in the actual flash player, it seems to just raise an error.
+        if offset + length > combining_bytes.len() {
+            return Err("EOFError: Reached EOF".into());
+        }
+        if let Some(this) = this {
+            if let Some(mut bytearray) = this.as_bytearray_mut(activation.context.gc_context) {
+                bytearray.write_bytes(if length != 0 {
+                    &combining_bytes[offset..length + offset]
+                } else {
+                    &combining_bytes[offset..]
+                });
+            }
+        }
+    }
+
+    Ok(Value::Undefined)
+}
+
+// Reads the bytes from the current bytearray into another bytearray
+pub fn read_bytes<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(this) = this {
+        let current_bytes = this
+            .as_bytearray_mut(activation.context.gc_context)
+            .unwrap()
+            .bytes()
+            .clone();
+        let position = this
+            .as_bytearray_mut(activation.context.gc_context)
+            .unwrap()
+            .position();
+        let mut merging_offset = 0;
+        if let Some(Value::Object(second_array)) = args.get(0) {
+            let offset = args
+                .get(1)
+                .unwrap_or(&Value::Unsigned(0))
+                .coerce_to_u32(activation)? as usize;
+            let length = args
+                .get(2)
+                .unwrap_or(&Value::Unsigned(0))
+                .coerce_to_u32(activation)? as usize;
+
+            if position + length > current_bytes.len() {
+                return Err("EOFError: Reached EOF".into());
+            }
+            if let Some(mut merging_storage) =
+                second_array.as_bytearray_mut(activation.context.gc_context)
+            {
+                // Offset should not be greater then the buffer
+                if merging_storage.bytes().len() < offset {
+                    return Ok(Value::Undefined);
+                }
+
+                let to_write = if length != 0 {
+                    current_bytes[position..length + position].to_vec()
+                } else {
+                    current_bytes[position..].to_vec()
```

Comment:
Don't think `to_vec` is necessary here, we can just use the slice directly: `&current_bytes[position..]` etc., particularly since the vec is already cloned at the start of the function.